src/tzdb.rs

```diff
@@ -252,6 +260,177 @@ impl Tzif {
         )
     }
 
+    pub fn get_named_tz_transition(
+        &self,
+        epoch_nanoseconds: i128,
+        direction: TransitionDirection,
+    ) -> TemporalResult<Option<EpochNanoseconds>> {
+        // First search the tzif data
+
+        let epoch_seconds = Seconds((epoch_nanoseconds / 1_000_000_000) as i64);
```

Comment:
thought(out-of-scope): It could be nice to use `Seconds` instead of nanoseconds or some other type than `i128`

This value could store the seconds value and an `is_exact` flag and have `from_seconds`, `from_milliseconds`, and `from_nanoseconds`, where the first two steps here -- the div_mod -- are part of the types ctors. It could help generalize the trait in comparison to i128.

Reply:
I don't think that would work, `is_exact` is messy to reason about in a generic context. If we want to we should just use ns here.